### ISSUE TITLE
VboxManage Error: Code E_ACCESSDENIED (0x80070005) - Access denied

### ISSUE THREAD

**Issue by Yohnah** [Yohnah-org/Docker#3]
When Yohnah/Docker is vagrant up, a message is raised up as follows:
~~~
VboxManage Error: Code E_ACCESSDENIED (0x80070005) - Access denied
~~~

It looks like a bug on Virtualbox version 6.1.28 for MacOS (although it also seems the linux OS are also affected)

When vagrant up command attempt to run the following VBoxManage command, the above error message is got:

~~~
VBoxManage hostonlyif ipconfig vboxnet0 --ip 172.28.128.1 --netmask 255.255.255.0
~~~

It is not a vagrant or Yohnah/Docker box issue. It is necessary to downgrade to 6.1.26 version as workaround.

Once your Virtualbox was downgraded to 6.1.26, Yohnah/Box will be able to raise again up.

A issue was open at https://www.virtualbox.org/ticket/20626 for Oracle Virtualbox bug tracker

**Comment by Yohnah** [comment]
According to ticket 20626 of Oracle Virtualbox bug tracker, the latest virtualbox version 6.1.28 imposes additional control on the addresses that can be set for host-only interfaces.

Reviewing the VirtualBox documentation https://www.virtualbox.org/manual/ch06.html#network_hostonly, it is written as:

~~~
On Linux, Mac OS X and Solaris Oracle VM VirtualBox will only allow IP addresses in 192.68.56.0/21 range to be assigned to host-only adapters. For IPv6 only link-local addresses are allowed. If other ranges are desired, they can be enabled by creating /etc/vbox/networks.conf and specifying allowed ranges there. For example, to allow 10.0.0.0/8 and 192.168.0.0/16 IPv4 ranges as well as 2001::/64 range put the following lines into /etc/vbox/networks.conf:

      * 10.0.0.0/8 192.168.0.0/16
      * 2001::/64
      
Lines starting with the hash # are ignored. Next example allows any addresses, effectively disabling range control:

      * 0.0.0.0/0 ::/0
      
If the file exists, but no ranges are specified in it, no addresses will be assigned to host-only adapters. The following example effectively disables all ranges:

      # No addresses are allowed for host-only adapters
~~~

Therefore, as vagrant attempt to setup the virtual network as 172.28.128.0/24 onto Virtualbox virtual networks, it is not allowed because the range is out of scope to 192.68.56.0/21

As workaround, you must run the following comands to file /etc/vbox/networks.conf (if folder does not exist, you have to create it).

~~~
$ cat <<EOF | sudo tee /etc/vbox/networks.conf                                                                                                                                                                                                                                                                                 
* 192.0.0.0/8
* 172.0.0.0/8
* 10.0.0.0/8
EOF
~~~


A ticket exists on vagrant bug tracker on github to tracking an oficial fix at https://github.com/hashicorp/vagrant/issues/12553

**Comment by Yohnah** [comment]
Hashicorp teams has already fixed it to setup 192.168.56.0/21 as default CIDR and no workaround is required anymore
